### ISSUE TITLE
Data table with checkmark column example: checkboxes aren't interactive

### ISSUE THREAD

**Issue by carmacleod** [carbon-design-system/carbon#7143]
Are the checkboxes supposed to be interactive in the [data table with checkmark column example](https://carbon-components-react.netlify.app/iframe.html?id=datatable--with-checkmark-columns)? I can't get them to work with mouse or keyboard.

**Comment by tw15egan** [comment]
Taking a look 🕵️

**Comment by joshblack** [comment]
Added a PR over in: https://github.com/carbon-design-system/carbon/pull/7175 👀
